### ISSUE TITLE
How to properly init completions?

### ISSUE THREAD

**Issue by mike-code** [mattmc3/zephyr#16]
Hi.

big fan of antidote here so I wanted to give your completions a try too. I have a few questions here

1. If I use `mattmc3/zephyr path:plugins/completion`, do I still need to call, the following?
``` 
autoload -Uz compinit
compinit
```

I'm asking this because antidote [autocompletion docs](https://getantidote.github.io/completions) suggest: "run autoload+compinit manually *or* use plugin". I understood it as, if I use plugin, I don't  need to run compinit myself anymore. Would that be correct? nb: if I don't run compinit in my zshrc, some autocompletion won't work (like the ones from `/usr/local/share/zsh/site-functions`).

2. If I use `mattmc3/zephyr path:plugins/completion`, I don't (shouldn't) have to use `zsh-users/zsh-completions`, right?
3. If I comment out `zsh-users/zsh-completions`, I get odd errors like `(eval):1: _nano: function definition file not found` (I've deleted `~/.zcompdump` just in case) -- do you an idea why could that be?

**Comment by mattmc3** [comment]
Hi @mike-code! Thank you for the kind words and for using antidote. Happy to answer your questions.

> If I use mattmc3/zephyr path:plugins/completion, do I still need to call [compinit]

Not be default, no. This plugins calls `compint` for you, unless you want to do it yourself. If you prefer to run it yourself, add the following zstyle customization:

```zsh
zstyle ':zephyr:plugin:completion' manual 'yes'
```

Then, you can call the `run-compinit` function that Zephyr defines for you whenever you want, or you can stick to the Zsh conventional way and run `autoload -Uz compinit; compinit`

> I'm asking this because antidote [autocompletion docs](https://getantidote.github.io/completions) suggest: "run autoload+compinit manually or use plugin"

Zephyr provides one such plugin. You could also use `belak/zsh-utils path:completion`, `sorin-ionescu/prezto path:modules/completion`, though obviously I have some bias towards Zephyr and will ensure that it works well for `antidote` users.

> If I use mattmc3/zephyr path:plugins/completion, I don't (shouldn't) have to use zsh-users/zsh-completions, right?

Actually, you should. Assuming there are completions defined there that you want/need. When it comes to Zsh completions, there are 3 things being handled:
1. Build out `fpath` with completions, as defined in `_completion` files. This is what `zsh-users/zsh-completions` does
2. Define how you want your completions to work. This is done by defining `zstyles`, and Zephyr does this.
3. Run compinit to initialize your completions from your `fpath`. Zephyr also does this.

> If I comment out zsh-users/zsh-completions, I get odd errors like (eval):1: _nano: function definition file not found (I've deleted ~/.zcompdump just in case) -- do you an idea why could that be?

That's usually due to your completions being cached, and `_nano` is defined by [zsh-completions/src/_nano](https://github.com/zsh-users/zsh-completions/blob/master/src/_nano).

TLDR; the recommendation for your .zsh_plugins.txt file for antidote might look like this:

```zsh
# ~/.zsh_plugins.txt
zsh-users/zsh-completions kind:fpath path:src
mattmc3/zephyr path:plugins/completion
```

**Comment by mattmc3** [comment]
I have pushed a new set of commits that I think addresses these sorts of completion questions once and for all. I have used hooks to add a new post_zshrc event. If you include Zephyr's completion plugin, it will queue up all calls to `compdef` for you and then it will run `compinit` for you at the end of your zshrc. You can also choose to run compinit yourself whenever you're ready, but Zephyr will now do it for you at the end of your config if you don't manually run it. This way, you can safely include completions at the top of your plugin list, continue to build out fpath with subsequent plugins, and be assured that your completions will pick up everything and apply any compdefs calls your plugins make. I've been testing this for about 6 months with great success, and feel it's ready for inclusion in Zephyr.
